presto-main/src/main/java/com/facebook/presto/split/BufferingSplitSource.java

```diff
@@ -45,6 +45,12 @@ public ConnectorId getConnectorId()
     }
 
     @Override
+    public Split getEmptySplit()
```

Comment:
The connectorId does not need to match a real connector id. The dispatching is based entirely on the split, so as long as PageSourceProvider knows how to deal with those special splits, it should just work.

In fact, you can probably just create a pseudo-connector (just the split source and the page source). Then, nothing other that the server setup and the scheduler need to know about the magic empty split source and empty splits.